retrievalmarket/impl/client.go

```diff
@@ -452,20 +459,32 @@ func (c *clientDealEnvironment) CloseDataTransfer(ctx context.Context, channelID
 	return err
 }
 
+// FinalizeBlockstore is called when all blocks have been received
+func (c *clientDealEnvironment) FinalizeBlockstore(ctx context.Context, dealID retrievalmarket.DealID) error {
```

Comment:
I see that we call tis function only on the happy path right now. We need to call this function even if the deal errors for whatever reason. If we don't do that, we will leak both the temp CARv2 file and the blockstore state in the store tracker.

Reply:
Basically, call this before the retrieval client enters any of the terminal/finality state,